grblas/formatting.py

```diff
@@ -202,18 +204,89 @@ def _get_matrix_dataframe(matrix, max_rows, min_rows, max_columns, *, mask=None)
                 column_offset,
                 mask=mask,
             )
+    if (
+        (mask is None or mask.structure)
+        and df.shape != matrix.shape
+        and min(matrix._nvals, max_rows if matrix._nvals <= max_rows else min_rows)
+        > 2 * df.count().sum()
+    ):
+        # The data is sparse and it's better to show in COO format.
+        # SS, SuiteSparse-specific: head
+        num_rows = matrix._nvals if matrix._nvals <= max_rows else min_rows
+        if matrix._is_transposed:
+            cols, rows, vals = matrix._matrix.ss.head(num_rows, sort=True)
```

Comment:
Note to self: sorting here for a better repr result.  If we find this is too slow for large objects, we can update it so we only sort for small objects (but we need to figure out where the threshold may be).